# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/test/mochitest/create_notification.html
# Commit: f9f3e107f36c
# Full Hash: f9f3e107f36c8cce438711ecef1b1a9574ae8f93
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2025-03-22 09:02:07
# Regressor Bug: 1948339
# File Overlap Count: 2
# Description:
#   Bug 1948339 - Part 2: Add NotificationObserver r=asuth
#   
#   This also makes page-bound notifications to not be stored via NotificationStorage.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D240642
# ==============================================================================

diff -r b84ed8a9da4c -r f9f3e107f36c dom/notification/test/mochitest/create_notification.html
--- a/dom/notification/test/mochitest/create_notification.html	Fri Mar 21 09:52:31 2025 +0000
+++ b/dom/notification/test/mochitest/create_notification.html	Fri Mar 21 09:52:31 2025 +0000
@@ -10,7 +10,10 @@
 // https://github.com/BenoitZugmeyer/eslint-plugin-html/issues/139
 (async () => {
   await NotificationTest.allowNotifications();
-  new Notification("This is a title", {
+
+  let swr = await navigator.serviceWorker.register("notification_empty_sw.js");
+  await navigator.serviceWorker.ready;
+  await swr.showNotification("This is a title", {
     body: "This is a notification body",
     tag: "sometag",
   });